Makefile

```diff
@@ -48,35 +48,57 @@ word-hyphen = $(word $2,$(subst -, ,$1))
 
 .EXPORT_ALL_VARIABLES:
 
-.PHONY: build-aws-openzfs-csi-driver
-build-aws-openzfs-csi-driver:
-	mkdir -p bin
-	CGO_ENABLED=0 GOOS=linux go build -ldflags ${LDFLAGS} -o bin/aws-fsx-openzfs-csi-driver ./cmd/
+.PHONY: linux/$(ARCH) bin/aws-fsx-openzfs-csi-driver
+linux/$(ARCH): bin/aws-fsx-openzfs-csi-driver
+bin/aws-fsx-openzfs-csi-driver: | bin
+	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) go build -mod=mod -ldflags ${LDFLAGS} -o bin/aws-fsx-openzfs-csi-driver ./cmd/
 
+# Builds all images
 .PHONY: all
 all: all-image-docker
 
+# Builds all images and pushes them
 .PHONY: all-push
-all-push:
-	docker buildx build \
-		--no-cache-filter=linux-amazon \
-		--platform=$(PLATFORM) \
-		--progress=plain \
-		--target=$(OS)-$(OSVERSION) \
-		--output=type=registry \
-		-t=$(IMAGE):$(TAG) \
-		.
-	touch $@
+all-push: all-image-registry push-manifest
```

Comment:
it seems like push-manifest already calls all-image-registry, I'm not convinced we need to call it again. Testing with the lustre csi driver

Reply:
Thank you for making this independent observation.